### ISSUE TITLE
Allow configuration of serializer_json

### ISSUE THREAD

**Issue by nuest** [trestletech/plumber#220]
In my app I am exposing arbitrary R objects as JSON, not for all of which the method `asJSON` works, e.g. objects of class `call`, getting the error

```
<simpleError: No method asJSON S3 class: call>
```

`asJSON(force = TRUE)` helps in so far as it returns an empty object for `call`.

It's simple enough for me to create my own serializer, but would you consider making the existing one configurable?

```
serializer_json <- function(...){
  function(val, req, res, errorHandler){
    tryCatch({
      json <- jsonlite::toJSON(val, ...)

      res$setHeader("Content-Type", "application/json")
      res$body <- json

      return(res$toResponse())
    }, error=function(e){
      errorHandler(req, res, e)
    })
  }
}

# not tested yet:
serializer_unboxed_json <- function(){
  serializer_json(auto_unbox = TRUE)
}
```

Happy to create a PR and add tests if you like the idea.

**Comment by trestletech** [comment]
Hm. I'm open to the idea.

My only hesitation there is that that pins us to using jsonlite's serialization moving forward. I doubt we'd be able to back that decision out anyways, since it's likely that we'd break folks' APIs in subtle ways.

Can you help me understand how you'd want to leverage this feature? Are you expecting that arbitrary arguments following `@serializer` would be passed in to the serializer function?

**Comment by nuest** [comment]
Yes, I would expect passing on to the serializer. Maybe using an explicit argument would be better?

```
serializer_json <- function(serializer_opts = c()){
  function(val, req, res, errorHandler){
    tryCatch({
      json <- jsonlite::toJSON(val, serializer_opts)
```

I have no experience with `...`, I must admit, but I can test this more if you're interested.

**Comment by jfdesomzee** [comment]
I have the same issue. Default parameters of toJSON function of jsonlite do not work for my case. I would need to have simplifyDataFrame = FALSE.

**Comment by ruiyiz** [comment]
I concur the request. In my case, I prefer to send data.frames as a dictionary of columns in json. But the default json serializer defaults to row-based data.frame serialization.

Similar to OP, it's easy enough to implement as a custom serializer, but would be great to have some ways to customize the behavior of the default json serializer.

**Comment by nuest** [comment]
@trestletech As for your hesitation for fixing a dependency to `jsonlite`: You mean you might want to keep the name and signature of `serializer_json(...)` but change the internal library used?

Maybe `serializer_json <- function(jsonlite_opts = c()){` is a way to go then. Other future JSON serializations might ignore this (you can deprecate/warn people using them) and get their own parameter `betterjson_opts` for passing in options. I am not sure it's worth the complexity though just to keep existing implementations working. That's what major releases are for.

**Comment by schloerke** [comment]
Fixed in #356
